### PR TITLE
Support Multiple NICs in VirtualMachine Status

### PR DESCRIPTION
**Pull request by sreyasn**
This change adds support for multiple network interfaces in
the VirtualMachine Status.

testing done:
make-all

### CODE REVIEW

File: api/v1alpha1/virtualmachine_types.go [hunk 1]
```diff
@@ -281,6 +281,20 @@ type VirtualMachineVolumeStatus struct {
 	Error string `json:"error"`
 }
 
+// NetworkInterfaceStatus defines the observed state of the network interfaces desired to be attached
```
Comment (janitha) [inline api/v1alpha1/virtualmachine_types.go:284]: I would reword this comment to clarify these are the interfaces as seen by vmtools and the guest os. 

(the desired interfaces would be configured in the spec, but it's possible the os would have ignored them, specially in future cases where additional images are supported)

File: api/v1alpha1/virtualmachine_types.go [hunk 2]
```diff
@@ -299,8 +313,7 @@ type VirtualMachineStatus struct {
 	// +optional
 	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
 
-	// VmIp describes the IP address of the VirtualMachine.  Currently, a VirtualMachine only supports a single
-	// network interface and interface address.
+	// VmIp describes the IP address of the VirtualMachine.
```
Comment (janitha) [inline api/v1alpha1/virtualmachine_types.go:316]: Possibly clarify that multiple IPs can be available for the VM, and to see the NetworkInterfaces for additional ones.

**Review by janitha — APPROVED** [review]

File: api/v1alpha1/virtualmachine_types.go (hunk 1 above)
Comment (sreyasn) [reply api/v1alpha1/virtualmachine_types.go:284]: sounds good. To verify, Is it okay for us to mention VMware tools in the comments here?.

Comment (janitha) [reply api/v1alpha1/virtualmachine_types.go:284]: Yeah, I don't see any problem with that

File: api/v1alpha1/virtualmachine_types.go (hunk 2 above)
Comment (bryanv) [reply api/v1alpha1/virtualmachine_types.go:316]: > Possibly clarify that multiple IPs can be available for the VM, and to see the NetworkInterfaces for additional ones.

The VMODL docs that correspond is 
> Primary IP address assigned to the guest operating system, if known.

I'd make some mention of that to help users better understand what this IP is vs the others.

File: api/v1alpha1/virtualmachine_types.go (hunk 1 above)
Comment (bryanv) [reply api/v1alpha1/virtualmachine_types.go]: I think later we'll have to incorporate and match these up with their corresponding spec entry when possible

**Review by bryanv — APPROVED** [review]
